src/codegen/properties.rs

```diff
@@ -83,28 +84,39 @@ fn declaration(env: &Env, prop: &Property) -> String {
         let param_type = if let Some(Bound {
             alias,
             ref type_str,
+            ref bound_type,
             ..
         }) = prop.bound
         {
-            let value_bound = if !prop.is_get {
+            if bound_type.is_into() {
+                let mut bounds = Bounds::default();
+                bounds.add_parameter(&prop.var_name,
+                                     &parameter_rust_type(env, prop.typ, dir, prop.nullable, prop.set_in_ref_mode).into_string(),
+                                     bound_type.clone(),
+                                     false);
+                bound = codegen::function::bounds(&bounds, &[], false);
+                format!("{}", bounds.iter().next().unwrap().alias)
```

Comment:
Alias from inner bounds used while bounds destroyed,
maybe just change to TYPE_PARAMETERS_START ?

Reply:
Ignore this, was wrong.

Reply:
May be this helps: fix for non-trait cases:
```diff
--- a/src/codegen/properties.rs
+++ b/src/codegen/properties.rs
@@ -1,5 +1,6 @@
 use std::io::{Result, Write};
 
+use analysis;
 use analysis::bounds::{Bound, Bounds};
 use analysis::properties::Property;
 use analysis::rust_type::{parameter_rust_type, rust_type};
@@ -89,9 +90,10 @@ fn declaration(env: &Env, prop: &Property) -> String {
         }) = prop.bound
         {
             if bound_type.is_into() {
+                let type_name = parameter_rust_type(env, prop.typ, dir, library::Nullable(false), analysis::ref_mode::RefMode::ByRefFake).into_string();
                 let mut bounds = Bounds::default();
                 bounds.add_parameter(&prop.var_name,
-                                     &parameter_rust_type(env, prop.typ, dir, prop.nullable, prop.set_in_ref_mode).into_string(),
+                                     &type_name,
                                      bound_type.clone(),
                                      false);
                 bound = codegen::function::bounds(&bounds, &[], false);
```